### DISCUSSION TITLE
Set_bucket example

### DISCUSSION THREAD

**Discussion by bLammT** [jeroenterheerdt/HAsmartirrigation#344] [Q&A]
Hey, sorry about this question. I probably don't know enough about HA in general. But am trying to get a smart irrigation system running and to do some testruns, I need to set the bucket to something except 0. 

Can you give an example how that service could be executed? Especially how to state the number of seconds. I tried everything around this example you could think of:
```
service: smart_irrigation.set_all_buckets
data: {
   states:'10'
   }
```

Thank you a lot!

**Comment by jeroenterheerdt** [comment]
something like this will work (as shown in the built-in example)
```
service: smart_irrigation.set_all_buckets
data:
  new_bucket_value: -10

```

![image](https://github.com/jeroenterheerdt/HAsmartirrigation/assets/8188990/858c3bb8-c443-4e63-bee2-ec70732e4214)

**Reply by bLammT** [reply]
Thank you for the fast reply. Seems like I need a little more help though.. I did set it up like this:
![image](https://github.com/jeroenterheerdt/HAsmartirrigation/assets/88188560/232cb1e3-b553-4608-ad36-39bdb5bbe4b6)

and this:
![image](https://github.com/jeroenterheerdt/HAsmartirrigation/assets/88188560/abe6abdf-28db-4505-912f-adfed298fbed)

and neither the bucket sensor, nor the sensor it self is showing something when i execute set bucket, neither when i calculate_all_zones afterwards. Something doesn't seem right or I just got the syntax wrong or smth :D 

[edit: updated screenshot]

**Reply by bLammT** [reply]
These are the cards for sensor.smart_irrigation[my_zone] and the "Bucket_Balkon" i created as tempalte. 

![image](https://github.com/jeroenterheerdt/HAsmartirrigation/assets/88188560/f2cab8aa-8b58-4e77-9a48-034c9b76c4ea)

The sensor is updating at 23:00 at night, like set, but i cannot manually change that through the mentioned comments to do some testing

**Reply by jeroenterheerdt** [reply]
Why are you setting the bucket manually in the first place? The bucket is set automatically by default (during calculation).

**Reply by bLammT** [reply]
I am trying to test my automations, since the first try didn't work.
I tried like this:
`alias: Balkonbewässerung_19:00
description: ""
trigger:
  - platform: event
    event_type: smart_irrigation_start_irrigation_all_zones
  - platform: time
    at: "19:00:00"
condition:
  - condition: numeric_state
    entity_id: sensor.smart_irrigation_balkon
    above: 0
    enabled: true
action:
  - type: turn_on
    device_id: 6a8bf0c535dcf89a5df261e4f95ba915
    entity_id: 56198a23dd7edf54c3bc6aab0ba326ee
    domain: switch
  - delay: "{{states(sensor.smart_irrigation_balkon)}}"
    enabled: true
  - type: turn_off
    device_id: 6a8bf0c535dcf89a5df261e4f95ba915
    entity_id: 56198a23dd7edf54c3bc6aab0ba326ee
    domain: switch
  - service: smart_irrigation.reset_all_buckets
    data: {}
mode: single`

It seems to just run infinitely (even though the duration on my entity is ~120)
When I tried around with other delays, the bucket was reset naturally, and since theres rain forecast every other day the last few weeks, I tried a workaround of setting my duration to something manually, to be able to test again..

But there will be better weather coming up, which I can use to play around further :)

**Reply by k-rock** [reply]
Hi - I am noticing that all of my buckets are stuck on 0s... do I need to setup something to trigger these to update? any other thoughts for why these might be stuck on 0s for my 3 zones?
